backend/msgpack/conn.go

```diff
@@ -76,12 +79,17 @@ func newConnection(addr string, opts ConnectionOptions) *connection {
 		maxThreshold:  opts.MaxReconnectThreshold(),
 		nowFn:         opts.ClockOptions().NowFn(),
 		threshold:     opts.InitReconnectThreshold(),
+		log:           opts.InstrumentOptions().Logger(),
 	}
 	c.connectWithLockFn = c.connectWithLock
 	c.writeWithLockFn = c.writeWithLock
 
 	c.Lock()
-	c.connectWithLockFn()
+	if err := c.connectWithLockFn(); err != nil {
+		c.log.WithFields(
```

Comment:
Do we really need to log here instead of emitting a metric? I worry if an agg tier host is down, the log messages are going to heavily pollute the proxycollector logs on every host. FWIW, the prod collector does not log this either.